src/features/fireWeather/components/TempRHGraph.tsx

```diff
@@ -0,0 +1,366 @@
+      const margin = { top: 10, right: 40, bottom: 40, left: 40 }
+      const widthValue = 600
+      const heightValue = 150
+      const width = widthValue - margin.left - margin.right
+      const height = heightValue - margin.top - margin.bottom
+      const svg = d3
+        .select(svgRef.current)
+        .attr('viewBox', `0 0 ${widthValue} ${heightValue}`)
+        .append('g')
+        .attr('transform', `translate(${margin.left}, ${margin.top})`)
+
+      /* Create scales for x and y axis */
+      const xScale = d3
+        .scaleTime()
+        .domain(minAndMaxDate)
+        .range([0, width])
+      const yTempScale = d3
+        .scaleLinear()
+        .domain([-10, 45])
+        .range([height, 0])
+      const yRHScale = d3
+        .scaleLinear()
+        .domain([0, 100])
+        .range([height, 0])
+
+      /* Render area and dots for temperature */
+      const historicModelTempArea = d3
+        .area<HistoricModel>()
+        .curve(d3.curveNatural)
+        .x(d => xScale(d.date))
+        .y0(d => yTempScale(d.tmp_tgl_2_90th))
+        .y1(d => yTempScale(d.tmp_tgl_2_5th))
+      svg
+        .append('path')
+        .datum(historicModels)
+        .attr('class', 'historicModelTempArea')
+        .attr('d', historicModelTempArea)
+        .attr('data-testid', 'historic-model-temp-area')
+      svg
+        .selectAll('.readingTempDot')
+        .data(readingValues)
+        .enter()
+        .append('circle')
+        .attr('class', 'readingTempDot')
+        .attr('cx', d => xScale(d.date))
+        .attr('cy', d => yTempScale(d.temp))
+        .attr('r', 1)
+        .attr('data-testid', 'wx-data-reading-temp-dot')
+      svg
+        .selectAll('.modelTempDot')
+        .data(modelValues)
+        .enter()
+        .append('circle')
+        .attr('class', 'modelTempDot')
+        .attr('cx', d => xScale(d.date))
+        .attr('cy', d => yTempScale(d.modelTemp))
+        .attr('r', 1)
+        .attr('data-testid', 'wx-data-model-temp-dot')
+
+      /* Render area and dots for RH */
+      const historicModelRHArea = d3
+        .area<HistoricModel>()
+        .curve(d3.curveNatural)
+        .x(d => xScale(d.date))
+        .y0(d => yRHScale(d.rh_tgl_2_90th))
+        .y1(d => yRHScale(d.rh_tgl_2_5th))
+      svg
+        .append('path')
+        .datum(historicModels)
+        .attr('class', 'historicModelRHArea')
+        .attr('d', historicModelRHArea)
+      svg
+        .selectAll('.readingRHDot')
+        .data(readingValues)
+        .enter()
+        .append('circle')
+        .attr('class', 'readingRHDot')
+        .attr('cx', d => xScale(d.date))
+        .attr('cy', d => yRHScale(d.rh))
+        .attr('r', 1)
+      svg
+        .selectAll('.modelRHDot')
+        .data(modelValues)
+        .enter()
+        .append('circle')
+        .attr('class', 'modelRHDot')
+        .attr('cx', d => xScale(d.date))
+        .attr('cy', d => yRHScale(d.modelRH))
+        .attr('r', 1)
```

Comment:
nice work. I can imagine this required a lot of patience to get working correctly